R/FilterStateChoices.R

```diff
@@ -292,12 +292,15 @@ ChoicesFilterState <- R6::R6Class( # nolint
                   updateCheckboxGroupInput(
                     session = session,
                     inputId = "selection",
+                    choiceNames = private$get_choice_labels(),
+                    choiceValues = as.character(private$choices),
                     selected = self$get_selected()
                   )
                 } else {
                   teal.widgets::updateOptionalSelectInput(
                     session = session,
                     inputId = "selection",
+                    choices = stats::setNames(private$choices, private$get_choice_labels()),
```

Comment:
what's the reason for removing choices from the updateCheckbox and adding it to updateOptionalSelect ? Similarly in the logical case below

Reply:
Hmm I didn't removed any choices. I just add everything available to updateXxx (choices, labels, selected) whenever update is run. I did it as I noticed [here](https://github.com/insightsengineering/teal.slice/pull/180/commits/d4f5acc18e20c344867265137fc7244c45787a8d) when I added 
`label = private$get_inf_label(),` to updateCheckboxInput call and it prevented  one reactive cycle thanks to this (label didn't need to recompute).

Reply:
Hmm I just realized that we have two observers to updateXyz - I will try to combine them into one

Reply:
ah yeah, my mistake